core/src/impl/Kokkos_Core.cpp

```diff
@@ -651,6 +651,140 @@ void pre_initialize_internal(const Kokkos::InitializationSettings& settings) {
+#elif defined(KOKKOS_ARCH_ARMV81)
+  declare_configuration_metadata("architecture", "CPU architecture", "ARMV81");
+#elif defined(KOKKOS_ARCH_ARMV8_THUNDERX)
+  declare_configuration_metadata("architecture", "CPU architecture",
+                                 "ARMV8_THUNDERX");
+#elif defined(KOKKOS_ARCH_ARMV8_THUNDERX2)
+  declare_configuration_metadata("architecture", "CPU architecture",
+                                 "ARMV8_THUNDERX2");
+#elif defined(KOKKOS_ARCH_BDW)
+  declare_configuration_metadata("architecture", "CPU architecture", "BDW");
+#elif defined(KOKKOS_ARCH_BGQ)
+  declare_configuration_metadata("architecture", "CPU architecture", "BGQ");
+#elif defined(KOKKOS_ARCH_HSW)
+  declare_configuration_metadata("architecture", "CPU architecture", "HSW");
+#elif defined(KOKKOS_ARCH_ICL)
+  declare_configuration_metadata("architecture", "CPU architecture", "ICL");
+#elif defined(KOKKOS_ARCH_ICX)
+  declare_configuration_metadata("architecture", "CPU architecture", "ICX");
+#elif defined(KOKKOS_ARCH_KNC)
+  declare_configuration_metadata("architecture", "CPU architecture", "KNC");
+#elif defined(KOKKOS_ARCH_KNL)
+  declare_configuration_metadata("architecture", "CPU architecture", "KNL");
+#elif defined(KOKKOS_ARCH_NATIVE)
+  declare_configuration_metadata("architecture", "CPU architecture", "NATIVE");
+#elif defined(KOKKOS_ARCH_POWER7)
+  declare_configuration_metadata("architecture", "CPU architecture", "POWER7");
+#elif defined(KOKKOS_ARCH_POWER8)
+  declare_configuration_metadata("architecture", "CPU architecture", "POWER8");
+#elif defined(KOKKOS_ARCH_POWER9)
+  declare_configuration_metadata("architecture", "CPU architecture", "POWER9");
+#elif defined(KOKKOS_ARCH_SKL)
+  declare_configuration_metadata("architecture", "CPU architecture", "SKL");
+#elif defined(KOKKOS_ARCH_SKX)
+  declare_configuration_metadata("architecture", "CPU architecture", "SKX");
+#elif defined(KOKKOS_ARCH_SNB)
+  declare_configuration_metadata("architecture", "CPU architecture", "SNB");
+#elif defined(KOKKOS_ARCH_SPR)
+  declare_configuration_metadata("architecture", "CPU architecture", "SPR");
+#elif defined(KOKKOS_ARCH_WSM)
+  declare_configuration_metadata("architecture", "CPU architecture", "WSM");
+#elif defined(KOKKOS_ARCH_AMD_ZEN)
+  declare_configuration_metadata("architecture", "CPU architecture", "AMD_ZEN");
+#elif defined(KOKKOS_ARCH_AMD_ZEN2)
+  declare_configuration_metadata("architecture", "CPU architecture",
+                                 "AMD_ZEN2");
+#elif defined(KOKKOS_ARCH_AMD_ZEN3)
+  declare_configuration_metadata("architecture", "CPU architecture",
+                                 "AMD_ZEN3");
+#else
+  declare_configuration_metadata("architecture", "CPU architecture", "none");
+#endif
+
+#if defined(KOKKOS_ARCH_INTEL_GEN)
+  declare_configuration_metadata("architecture", "GPU architecture",
+                                 "INTEL_GEN");
+#elif defined(KOKKOS_ARCH_INTEL_DG1)
```

Comment:
Can there ever be a case where `if` should be used instead of `elif`? (multiple architectures?)

Reply:
No, only one GPU architecture can be used.